src/input/reader_to_storage.cc

```diff
@@ -373,6 +429,9 @@ StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::Selection *
 		int value = selection->name_to_int( item_name  );
 		return new StorageInt( value );
 	} catch (ExcInputError & e) {
+		if (try_transpose_read_) {
```

Comment:
Exceptions should be used only for errors. So we have to check for the array on input just before we call get_string_value().